src/urllib3/exceptions.py

```diff
@@ -160,10 +161,14 @@ class ConnectTimeoutError(TimeoutError):
     pass
 
 
-class NewConnectionError(ConnectTimeoutError, PoolError):
+class NewConnectionError(ConnectTimeoutError, HTTPError):
     """Raised when we fail to establish a new connection. Usually ECONNREFUSED."""
 
-    pass
+    pool: "SOCKSConnection"
```

Comment:
I think you'll have to use the parent class `HTTPConnection` when typing connectionpool.py, so let's use that right away?

It's also a bit weird to use `pool` instead of `conn` but I guess changing that would be a breaking change

Reply:
Could change to `conn` and add a deprecated property `pool`?

Reply:
Since the exception is no longer a PoolError I would go ahead and switch to conn. Let's just document that as part of the 2.0 breaking changes.

This is the kind of subtle issues that get fixed when you add types :)